src/dc_receive_imf.c

```diff
@@ -1267,15 +1269,34 @@ void dc_receive_imf(dc_context_t* context, const char* imf_raw_not_terminated, s
 				}
 			}
 
+			if ((field=dc_mimeparser_lookup_field(mime_parser, "In-Reply-To"))!=NULL
+			 && field->fld_type==MAILIMF_FIELD_IN_REPLY_TO)
+			{
+				struct mailimf_in_reply_to* fld_in_reply_to = field->fld_data.fld_in_reply_to;
+				if (fld_in_reply_to) {
+					mime_in_reply_to = dc_str_from_clist(field->fld_data.fld_in_reply_to->mid_list, " ");
```

Comment:
10-20 lines above you save mime headers -- is it correct to set mime_in_reply_to afterwards instead of before?

Reply:
this should be fine.

- dc_str_from_clist() reads the header from the hashed-cache to find out the current value. no headers are modified here
- "10-20 lines above" we tear off the header from the original message to allow parsing and testing from python - this is also only a reading process

so, both will contain the same headers.
the writing to db happens a few lines later then.